docs/src/content/docs/concepts/arrays.mdx

```diff
@@ -0,0 +1,202 @@
+
+- Evaluating the usefulness of each chunk returned by your vector database
+- Translating a single document into multiple languages
+- Analyzing key themes in a set of customer reviews
+
+While you can accomplish this with standard GenSX components, it can lead to your components becoming more tightly coupled and less reusable. To solve this, GenSX supports `gsx.array` which allows you to chain together array operations like `map`, `filter`, `flatMap`, and `reduce`. The API follows the same conventions as arrays in JavaScript, making it familiar and intuitive to use.
+
+## Example
+
+Here's a practical example of using `gsx.array`. The `Research` component below receives a prompt and a list of queries, and returns a list of summaries from relevant research papers.
+
+```tsx
+const Research = gsx.Component<ResearchProps, ArxivSummary[]>(
+  "Research",
+  async ({ queries: string[], prompt: string }) => {
+    return await gsx
+      .array(queries)
+      .flatMap((query) => <ArxivSearch query={query} maxResults={3} />)
+      .filter((document) => (
+        <GradeDocument prompt={prompt} document={document} />
+      ))
+      .map((document) => (
+        <FetchAndSummarize document={document} prompt={prompt} />
+      ))
+      .toArray();
+  },
+);
+```
+
+Walking through this step by step, it:
+
+1. Instantiates a `GsxArray<string>` from the `queries`
+2. Uses `flatMap` to call the `ArxivSearch` component for each query and flatten the output into a single array of documents
+3. Calls `filter` to filter out any documents that an LLM judge deems irrelevant. `GradeDocument` returns a boolean
+4. Chains another `map` operation to call the `FetchAndSummarize` component for each document
+5. Calls `toArray()` to convert the `GsxArray<ArxivSummary>` into a `ArxivSummary[]`
+
+The full code is available in the [Deep Research example](https://github.com/gensx-inc/gensx/tree/main/examples/).
+
+## Working with arrays
+
+Now that you've seen an example showing a lot of the functionality of `gsx.array`, let's explore each of these operations in more detail.
+
+### Creating an array
+
+To access the array operations, you first need to create a `GsxArray`. Here's how to create an array from raw values:
+
+```tsx
+import { gsx } from "gensx";
+
+// Create from raw values
+const numbers = gsx.array([1, 2, 3]);
+```
+
+You can also create arrays from components:
+
+```tsx
+const NumberWrapper = gsx.Component<number, number>(
+  "NumberWrapper",
+  ({ value }) => value,
+);
+
+// Create from components
+const wrappedNumbers = gsx.array([
+  <NumberWrapper n={1} />,
+  <NumberWrapper n={2} />,
+  <NumberWrapper n={3} />,
+]);
+```
+
+This shows a simplistic example of creating an array from components but in practice you can do this with any component, no matter how complex.
+
+To convert a `GsxArray` back to a plain array, you simply call the `toArray()` method.
```

Comment:
should probably mention that `toArray` will produce an array of resolved component outputs.

Reply:
i like it. adding.